uaclient/messages.py

```diff
@@ -0,0 +1,674 @@
+from uaclient.defaults import BASE_UA_URL, DOCUMENTATION_URL
+
+
+class NamedMessage:
+    def __init__(self, name: str, msg: str):
+        self.name = name
+        self.msg = msg
+
+
+class FormatedNamedMessage(NamedMessage):
+    def __init__(self, name: str, msg: str):
+        self.name = name
+        self.tmpl_msg = msg
+
+    def format(self, **msg_params):
+        self.msg = self.tmpl_msg.format(**msg_params)
+
+        return self
```

Comment:
Doesn't matter that much, but a slightly different pattern to consider here (that I think is compatible with how this is used everywhere) might look like this:
```suggestion
class FormattedNamedMessage:
    def __init__(self, name: str, tmpl_msg: str):
        self.name = name
        self.tmpl_msg = tmpl_msg

    def format(self, **msg_params) -> NamedMessage:
        return NamedMessage(self.name, self.tmpl_msg.format(**msg_params))
```

Not worth it if it requires changing all the places where this is used though

Reply:
Does it make sense to just add `format` to `NamedMessage`? It would reduce complexity (and I think it is ok to format a message, semantically speaking) but not sure about how clear you want to be about the need to provide format args